flink-runtime/src/main/java/org/apache/flink/runtime/state/filesystem/FsCheckpointStreamFactory.java

```diff
@@ -297,7 +296,7 @@ public StreamStateHandle closeAndGetHandle() throws IOException {
 					if (outStream == null && pos <= localStateThreshold) {
 						closed = true;
 						byte[] bytes = Arrays.copyOf(writeBuffer, pos);
-						return new ByteStreamStateHandle(bytes);
+						return new ByteStreamStateHandle(String.valueOf(createStatePath()), bytes);
```

Comment:
Why don't we do `createStatePath().toString()`? We actually don't want a non-unique `"null"` String here in case that `createStatePath()` returns `null` (which it shouldn't anyways).